frontend/src/app/login-form/login-form.component.html

```diff
@@ -0,0 +1,42 @@
+<div class="login-container">
+  <mat-card class="login-card">
+    <mat-card-header class="login-header">
+      <div mat-card-avatar class="login-card-icon"></div>
+      <mat-card-title>Sign In</mat-card-title>
+    </mat-card-header>
+    <mat-card-content class="login-card-content">
+      <mat-form-field appearance="fill">
+        <mat-label>Username</mat-label>
+        <input matInput type="text" value="ala" #username />
+      </mat-form-field>
+      <mat-form-field appearance="fill">
+        <mat-label>Password</mat-label>
+        <input
+          matInput
+          [type]="hide ? 'password' : 'text'"
+          value="makota"
+          #password
+        />
+        <button
+          mat-icon-button
+          matSuffix
+          (click)="hide = !hide"
+          [attr.aria-label]="'Hide password'"
+          [attr.aria-pressed]="hide"
+        >
+          <mat-icon>{{ hide ? 'visibility_off' : 'visibility' }}</mat-icon>
+        </button>
+      </mat-form-field>
+    </mat-card-content>
+    <mat-card-actions>
+      <button
+        mat-raised-button
+        color="primary"
+        (click)="loginHandler(username.value, password.value)"
```

Comment:
Interesting approach. Why not `ngModel` on inputs?

Reply:
I don't see any reason to keep these values in `.ts` file, I simply pass them into `httpService` and do not modify them in any way in component.
Your approach is to call `loginHandler()` without any params and call `httpService` with `this.username` and `this.password`?

Reply:
Yeah, this piece of code is quite simple so it's readable yet it's not scaling well, I reckon. Setting the component level variable is better if you've got multiple places across the component which need to access the model variable.
In addition, using `ngModel` pattern makes the code easier to migrate to Angular's reactive forms which I personally found a way better method to manage state of forms.

Reply:
As I mentioned on Discord, I treat this app as a testing field for learning many ways of solving the same problem. In simple cases like this one (one usage of value, no modifications), I made a decision to use the simplest approach.
Thank you, for suggesting another one (maybe more complex). When the app grows, I surely will use it ;)